backend/dataall/modules/omics/services/omics_service.py

```diff
@@ -0,0 +1,185 @@
+"""
+A service layer for Omics pipelines
+Central part for working with Omics workflow runs
+"""
+
+import dataclasses
+import logging
+from dataclasses import dataclass, field
+from typing import List, Dict
+
+
+from dataall.base.context import get_context
+from dataall.core.environment.services.environment_service import EnvironmentService
+from dataall.core.permissions.services.resource_policy_service import ResourcePolicyService
+from dataall.core.permissions.services.tenant_policy_service import TenantPolicyService
+from dataall.core.permissions.services.group_policy_service import GroupPolicyService
+from dataall.modules.s3_datasets.db.dataset_repositories import DatasetRepository
+from dataall.base.db import exceptions
+import json
+
+from dataall.modules.omics.db.omics_repository import OmicsRepository
+from dataall.modules.omics.aws.omics_client import OmicsClient
+from dataall.modules.omics.db.omics_models import OmicsRun
+from dataall.modules.omics.services.omics_permissions import (
+    MANAGE_OMICS_RUNS,
+    CREATE_OMICS_RUN,
+    OMICS_RUN_ALL,
+    DELETE_OMICS_RUN,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class OmicsRunCreationRequest:
```

Comment:
is this dataclass used? Should it be passed to `create_omics_run()` ?

Reply:
Replaced data class by more checks in this function: def validate_creation_request(data)